runtime/src/main/java/com/flipkart/flux/controller/WorkFlowExecutionController.java

```diff
@@ -393,7 +393,12 @@ public void unsidelineState(String stateMachineId, Long stateId) {
             throw new IllegalStateException("State with the asked id: " + stateId + " not found in stateMachine with id: " + stateMachineId);
         }
 
-        if (askedState.getStatus() == Status.sidelined || askedState.getStatus() == Status.errored) {
+        Set<State> checkExecutableState = new HashSet<>();
```

Comment:
Make the parent method non void, send response accordingly if the state is not eligible for un-sidelining.

Reply:
Done. added appropriate response codes.

Reply:
No i meant to raise exception in workflowExecutionController, and send appropriate response from ApiResourceClass.

Reply:
Done.